File: clients/node/sdk/models/federationUsage.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.FederationUsage = void 0;
class FederationUsage {
    /**
     * Gets the active property value. The active property
     * @returns a int64
     */
    get active() {
        return this._active;
    }
    ;
    /**
     * Sets the active property value. The active property
     * @param value Value to set for the active property.
     */
    set active(value) {
        this._active = value;
    }
    ;
    /**
     * Gets the additionalData property value. Stores additional data not described in the OpenAPI description found when deserializing. Can be used for serialization as well.
     * @returns a Record<string, unknown>
     */
    get additionalData() {
        return this._additionalData;
    }
    ;
    /**
     * Sets the additionalData property value. Stores additional data not described in the OpenAPI description found when deserializing. Can be used for serialization as well.
     * @param value Value to set for the AdditionalData property.
     */
    set additionalData(value) {
        this._additionalData = value;
    }
    ;
    /**
     * Instantiates a new FederationUsage and sets the default values.
     */
    constructor() {
        this._additionalData = {};
    }
    ;
    /**
     * The deserialization information for the current model
     * @returns a Record<string, (node: ParseNode) => void>
     */
    getFieldDeserializers() {
        return {
            "active": n => { this.active = n.getNumberValue(); },
            "receiving": n => { this.receiving = n.getNumberValue(); },
            "sending": n => { this.sending = n.getNumberValue(); },
        };
    }
    ;
    /**
     * Gets the receiving property value. The receiving property
     * @returns a int64
     */
    get receiving() {
        return this._receiving;
    }
    ;
    /**
     * Sets the receiving property value. The receiving property
     * @param value Value to set for the receiving property.
     */
    set receiving(value) {
        this._receiving = value;
    }
    ;
    /**
     * Gets the sending property value. The sending property
     * @returns a int64
     */
    get sending() {
        return this._sending;
    }
    ;
    /**
     * Sets the sending property value. The sending property
     * @param value Value to set for the sending property.
     */
    set sending(value) {
        this._sending = value;
    }
    ;
    /**
     * Serializes information the current object
     * @param writer Serialization writer to use to serialize this model
     */
    serialize(writer) {
        if (!writer)
            throw new Error("writer cannot be undefined");
        writer.writeNumberValue("active", this.active);
        writer.writeNumberValue("receiving", this.receiving);
        writer.writeNumberValue("sending", this.sending);
        writer.writeAdditionalData(this.additionalData);
    }
    ;
}
exports.FederationUsage = FederationUsage;

File: clients/csharp/sdk/Models/FederationUsage.cs
using Microsoft.Kiota.Abstractions.Serialization;
using System.Collections.Generic;
using System.IO;
using System.Linq;
using System;
namespace MastodonClientLib.Models {
    public class FederationUsage : IParsable {
        /// <summary>The active property</summary>
        public long? Active { get; set; }
        /// <summary>The receiving property</summary>
        public long? Receiving { get; set; }
        /// <summary>The sending property</summary>
        public long? Sending { get; set; }
        /// <summary>
        /// Creates a new instance of the appropriate class based on discriminator value
        /// </summary>
        /// <param name="parseNode">The parse node to use to read the discriminator value and create the object</param>
        public static FederationUsage CreateFromDiscriminatorValue(IParseNode parseNode) {
            _ = parseNode ?? throw new ArgumentNullException(nameof(parseNode));
            return new FederationUsage();
        }
        /// <summary>
        /// The deserialization information for the current model
        /// </summary>
        public IDictionary<string, Action<IParseNode>> GetFieldDeserializers() {
            return new Dictionary<string, Action<IParseNode>> {
                {"active", n => { Active = n.GetLongValue(); } },
                {"receiving", n => { Receiving = n.GetLongValue(); } },
                {"sending", n => { Sending = n.GetLongValue(); } },
            };
        }
        /// <summary>
        /// Serializes information the current object
        /// </summary>
        /// <param name="writer">Serialization writer to use to serialize this model</param>
        public void Serialize(ISerializationWriter writer) {
            _ = writer ?? throw new ArgumentNullException(nameof(writer));
            writer.WriteLongValue("active", Active);
            writer.WriteLongValue("receiving", Receiving);
            writer.WriteLongValue("sending", Sending);
        }
    }
}
